src/components/LivePreviewerComponents/RichTextEditor.js

```diff
@@ -0,0 +1,183 @@
+import React from "react";
+
+import {
+  Editor,
+  EditorState,
+  RichUtils,
+  getDefaultKeyBinding,
+  convertFromRaw,
+  convertToRaw,
+} from "draft-js";
+import "draft-js/dist/Draft.css";
+
+const RichTextEditor = (props) => {
+  const editorRef = React.useRef();
+  let editor;
+
+  try {
+    editor = convertFromRaw(JSON.parse(props.value));
+  } catch (e) {
+    editor = null;
+  }
+  const [editorState, setEditorState] = React.useState(() =>
+    !!editor ? EditorState.createWithContent(editor) : EditorState.createEmpty()
+  );
+
+  const onChange = (newEditorState) => {
+    props.onChange(JSON.stringify(convertToRaw(newEditorState.getCurrentContent())));
+    setEditorState(newEditorState);
+  };
+
+  const focus = () => editorRef?.current?.focus();
+
+  const _handleKeyCommand = (command, editorState) => {
+    const newState = RichUtils.handleKeyCommand(editorState, command);
+    if (newState) {
+      onChange(newState);
+      return true;
+    }
+    return false;
+  };
+
+  const _mapKeyToEditorCommand = (e) => {
+    if (e.keyCode === 9 /* TAB */) {
+      const newEditorState = RichUtils.onTab(e, editorState, 4 /* maxDepth */);
+      if (newEditorState !== editorState) {
+        onChange(newEditorState);
+      }
+      return;
+    }
+    return getDefaultKeyBinding(e);
+  };
+
+  const _toggleBlockType = (blockType) => {
+    onChange(RichUtils.toggleBlockType(editorState, blockType));
+  };
+
+  const _toggleInlineStyle = (inlineStyle) => {
+    onChange(RichUtils.toggleInlineStyle(editorState, inlineStyle));
+  };
+  let className = "RichEditor-editor";
+  const contentState = editorState.getCurrentContent();
+  if (!contentState.hasText()) {
+    if (contentState.getBlockMap().first().getType() !== "unstyled") {
+      className += " RichEditor-hidePlaceholder";
+    }
+  }
+
+  return (
+    <div className="RichEditor-root">
+      <BlockStyleControls editorState={editorState} onToggle={_toggleBlockType} />
+      <InlineStyleControls editorState={editorState} onToggle={_toggleInlineStyle} />
+      <div className={className} onClick={focus}>
+        <Editor
+          blockStyleFn={getBlockStyle}
+          customStyleMap={styleMap}
+          editorState={editorState}
+          handleKeyCommand={_handleKeyCommand}
+          keyBindingFn={_mapKeyToEditorCommand}
+          onChange={onChange}
+          placeholder="Description..."
+          ref={editorRef}
+          spellCheck={true}
+        />
+      </div>
+    </div>
+  );
+};
+
+const styleMap = {
+  CODE: {
+    backgroundColor: "rgba(0, 0, 0, 0.05)",
+    fontFamily: '"Inconsolata", "Menlo", "Consolas", monospace',
+    fontSize: 16,
+    padding: 2,
+  },
+};
+
+function getBlockStyle(block) {
+  switch (block.getType()) {
```

Comment:
I'm okay with leaving this here for the future, but this switch-case is not yet necessary here

```suggestion
  if (block.getType() === "blockquote") {
     return "RichEditor-blockquote";
  }
  
  return null;
```